packages/text-decorator/test/decorate-react.test.tsx

```diff
@@ -0,0 +1,251 @@
+    expect(decorateReact(0, kNullOptions)).toBe(0);
+    expect(decorateReact(1, kNullOptions)).toBe(1);
+    expect(decorateReact(false, kNullOptions)).toBe(false);
+    expect(decorateReact(true, kNullOptions)).toBe(true);
+  });
+
+  it("substitutes simple strings", () => {
+    const someOtherOptions: IDecorateReactOptions = { words: ['Some'], replace: 'Other' };
+    const textDataOptions: IDecorateReactOptions = { words: ['text'], replace: 'data' };
+    expect(decorateReact("Some Text", kNullOptions)).toBe("Some Text");
+    expect(decorateReact("Some Text", someOtherOptions)).toBe("Other Text");
+    expect(decorateReact("Text TEXT text", textDataOptions)).toBe("data data data");
+  });
+
+  it("ignores unknown objects", () => {
+    const empty = {};
+    expect(decorateReact(empty, kNullOptions)).toBe(empty);
+  });
+
+  it("handles arrays of simple values", () => {
+    const input = [undefined, null, 0, 1, false, true, "Some Text"];
+    const someMoreOptions: IDecorateReactOptions = { words: ['Some'], replace: 'More' };
+    const result = decorateReact(input, someMoreOptions);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(input).replace("Some", "More"));
+  });
+
+  it("works with compiled JSX/TSX", () => {
+    const input = <div></div>;
+    const options = { words: ['foo'], replace: '' };
+    const result = decorateReact(input, options);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(input));
+  });
+
+  it("works with compiled JSX/TSX", () => {
+    const input = <div>bar</div>;
+    const options = { words: ['foo'], replace: '' };
+    const result = decorateReact(input, options);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(input));
+  });
+
+  it("works with compiled JSX/TSX", () => {
+    const input = <div>Some <span>Text</span></div>;
+    const result = decorateReact(input, kNullOptions);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(input));
+  });
+
+  it("replaces text in compiled JSX/TSX", () => {
+    const input = <div>{["Some", "Text"]}</div>;
+    const someMoreOptions: IDecorateReactOptions = { words: ['Some'], replace: 'More' };
+    const result = decorateReact(input, someMoreOptions);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(input).replace('Some', 'More'));
+  });
+
+  it("replaces text, not tags in compiled JSX/TSX", () => {
+    const input = <div><span>div</span> span</div>;
+    const spanTextOptions: IDecorateReactOptions = { words: ['span'], replace: 'text' };
+    const expected = <div><span>div</span> text</div>;
+    const result = decorateReact(input, spanTextOptions) as ReactElement;
+    expect(JSON.stringify(result)).toBe(JSON.stringify(expected));
+  });
+
+  it("wraps complete text nodes with <span> nodes", () => {
+    const input = 'word';
+    const options: IDecorateReactOptions = {
+            words: ['word'],
+            replace: <span></span>
+          };
+    const expected = <span key='word-0'>word</span>;
```

Comment:
I guess I am surprised by this expectation.  Makes me worried I am missing something.

I would have thought that the expected result would be simply `<span></span>` because I didn't see a substitution in the replacement....

Reply:
This is making use of the convenience feature I mentioned above of assuming that the client wants to wrap the original string if no `$1` replacement was specified. At the time I was thinking that it made the API more convenient and less error-prone, but I can also see the argument that it's an unnecessary complication. I'm curious what you think having read the description of the intent.

Reply:
OK, I see it now!

Those are some sensible conveniences, but it wasn't intuitive to me (obviously).

I think its reasonable now that I have been clued in.

Thanks

- n

Reply:
But obviously if its well documented  -- no harm done.